sourced/ml/models/model_converters/merge_bow.py

```diff
@@ -0,0 +1,69 @@
+import os
+from scipy.sparse import hstack, vstack
+
+from sourced.ml import extractors
+from sourced.ml.models.bow import BOW
+from sourced.ml.models.model_converters.base import Model2Base
+
+
+class MergeBOW(Model2Base):
+    """
+    Merges several :class:`DocumentFrequencies` models together.
+    """
+    MODEL_FROM_CLASS = BOW
+    MODEL_TO_CLASS = BOW
+
+    def __init__(self, features=None, *args, **kwargs):
+        super().__init__(num_processes=1, *args, **kwargs)
+        self.documents = None
+        self.tokens = None
+        self.matrix = None
+        self.deps = None
+        self.features = None
+        if features:
+            self.features = [ex.NAMESPACE for ex in extractors.__extractors__.values()
+                             if ex.NAME in features]
+
+    def convert_model(self, model: BOW) -> None:
+        if self.tokens is None:
+            self.tokens = model.tokens
+            self.documents = model.documents
+            self.matrix = [model.matrix.tocsr()]
+            self.deps = model._meta["dependencies"]
+        elif set(self.tokens) != set(model.tokens):
+            raise ValueError("Models don't share the same set of tokens !")
+        else:
+            self.documents += model.documents
+            self.matrix.append(model.matrix.tocsr())
+
+    def finalize(self, index: int, destdir: str):
+        self._log.info("Stacking matrices ...")
+        matrix = self.matrix[0]
+        self.matrix.pop(0)
+        while self.matrix:
+            matrix = vstack([matrix, self.matrix[0]])
+            del self.matrix[0]
+            self._log.info("%s matrices to stack ...", len(self.matrix))
+        self.matrix = matrix
+        self._log.info("Writing model ...")
+        if self.features:
+            self._reduce_matrix()
+        BOW(log_level=self._log.level) \
+            .construct(self.documents, self.tokens, self.matrix) \
+            .save(self._save_path(index, destdir), self.deps)
+
+    def _reduce_matrix(self):
+        reduced_tokens = []
+        columns = []
+        for i, token in enumerate(self.tokens):
+            if token.split(".")[0] in self.features:
+                reduced_tokens.append(token)
+                columns.append(self.matrix.getcol(i).tocsc())
+        self.tokens = reduced_tokens
+        self.matrix = hstack(columns)
```

Comment:
I think It is better to convert `self.matrix` to csc first 
(`matrix = self.matrix.tocsc()`), 
than in loop  ~`columns.append(self.matrix.getcol(i).tocsc())`~ `columns.append(i)` 
and here ~`self.matrix = hstack(columns)`~ `self.matrix = matrix[columns]`